src/hooks.test.tsx

```diff
@@ -1054,3 +1054,44 @@ describe('useAPICache', () => {
     });
   });
 });
+
+test('passing a function for `client` is supported', async () => {
+  const TestContext = React.createContext<AxiosInstance | undefined>(undefined);
+
+  const { useAPIQuery } = createAPIHooks<TestEndpoints>({
+    name: 'test-name',
+    client: () => {
+      const client = React.useContext(TestContext);
+      if (!client) {
+        throw new Error('no client specified');
+      }
+      return client;
+    },
```

Comment:
this is one key use case that deserves document

Reply:
@shawnzhu Do you think the documentation I'm adding in this PR is clear enough? Or, should we add more.

Reply:
it is enough for this PR.

my above comment is really about the use case (e.g., use it in `phc-ui`), e.g., https://tanstack.com/query/v3/docs/react/guides/default-query-function

but it is more about optimization (for audience), could be done over time once user like me picks up this new feature